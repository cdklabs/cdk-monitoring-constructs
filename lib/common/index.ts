export * from "./alarm";
export * from "./metric";
export * from "./monitoring";
export * from "./url";
export * from "./widget";
export * from "./strings";
