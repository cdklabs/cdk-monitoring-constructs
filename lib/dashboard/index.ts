export * from "./BitmapDashboard";
export * from "./DashboardRenderingPreference";
export * from "./DynamicDashboardSegment";
export * from "./DashboardSegment";
export * from "./DashboardWithBitmapCopy";
export * from "./DefaultDashboardFactory";
export * from "./DynamicDashboardFactory";
export * from "./DefaultWidgetFactory";
export * from "./IDashboardFactory";
export * from "./IDynamicDashboardFactory";
export * from "./IWidgetFactory";
export * from "./MonitoringNamingStrategy";
export * from "./SingleWidgetDashboardSegment";
export * from "./widget";
