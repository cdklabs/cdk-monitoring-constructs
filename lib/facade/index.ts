export * from "./IMonitoringAspect";
export * from "./MonitoringFacade";
export * from "./MonitoringAspect";
