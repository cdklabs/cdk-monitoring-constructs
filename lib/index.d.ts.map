{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,UAAU,CAAC;AACzB,cAAc,aAAa,CAAC;AAC5B,cAAc,UAAU,CAAC;AACzB,cAAc,cAAc,CAAC"}