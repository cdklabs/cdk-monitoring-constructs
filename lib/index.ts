export * from "./common";
export * from "./dashboard";
export * from "./facade";
export * from "./monitoring";
