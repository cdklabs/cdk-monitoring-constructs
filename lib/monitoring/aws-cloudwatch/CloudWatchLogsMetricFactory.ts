import { DimensionsMap } from "aws-cdk-lib/aws-cloudwatch";

import {
  BaseMetricFactory,
  BaseMetricFactoryProps,
  MetricFactory,
  MetricStatistic,
} from "../../common";

const CloudWatchLogsNamespace = "AWS/Logs";

export interface CloudWatchLogsMetricFactoryProps
  extends BaseMetricFactoryProps {
  /**
   * Name of the log group to monitor.
   */
  readonly logGroupName: string;
}

export class CloudWatchLogsMetricFactory extends BaseMetricFactory<CloudWatchLogsMetricFactoryProps> {
  private readonly dimensionsMap: DimensionsMap;

  constructor(
    metricFactory: MetricFactory,
    props: CloudWatchLogsMetricFactoryProps,
  ) {
    super(metricFactory, props);

    this.dimensionsMap = {
      LogGroupName: props.logGroupName,
    };
  }

  metricIncomingLogEvents() {
    return this.metricFactory.createMetric(
      "IncomingLogEvents",
      MetricStatistic.N,
      "Logs",
      this.dimensionsMap,
      undefined,
      CloudWatchLogsNamespace,
      undefined,
      this.region,
      this.account,
    );
  }
}
