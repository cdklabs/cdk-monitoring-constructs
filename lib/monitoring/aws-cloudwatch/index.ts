export * from "./CloudWatchLogsMetricFactory";
export * from "./LogMonitoring";
