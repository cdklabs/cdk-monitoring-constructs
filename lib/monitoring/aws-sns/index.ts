export * from "./SnsTopicMetricFactory";
export * from "./SnsTopicMonitoring";
