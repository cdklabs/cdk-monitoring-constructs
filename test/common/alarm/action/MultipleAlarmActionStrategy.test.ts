import { Stack } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { Alarm, Metric } from "aws-cdk-lib/aws-cloudwatch";
import { Topic } from "aws-cdk-lib/aws-sns";

import { multipleActions, SnsAlarmActionStrategy } from "../../../../lib";

test("snapshot test: multiple actions", () => {
  const stack = new Stack();
  const topic1 = new Topic(stack, "DummyTopic1");
  const topic2 = new Topic(stack, "DummyTopic2");
  const topic3 = new Topic(stack, "DummyTopic3");
  const alarm = new Alarm(stack, "DummyAlarm", {
    evaluationPeriods: 1,
    threshold: 0,
    metric: new Metric({ namespace: "Dummy", metricName: "Dummy" }),
  });
  const action = multipleActions(
    new SnsAlarmActionStrategy({ onAlarmTopic: topic1 }),
    new SnsAlarmActionStrategy({ onAlarmTopic: topic2 }),
    new SnsAlarmActionStrategy({ onAlarmTopic: topic3 })
  );
  action.addAlarmActions({ alarm, action });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});
