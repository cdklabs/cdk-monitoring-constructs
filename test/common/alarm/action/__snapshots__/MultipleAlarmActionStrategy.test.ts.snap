// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: multiple actions 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "DummyAlarm234203A9": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "DummyTopic18541D2D0",
          },
          Object {
            "Ref": "DummyTopic2F00C887A",
          },
          Object {
            "Ref": "DummyTopic389AA825B",
          },
        ],
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "EvaluationPeriods": 1,
        "MetricName": "Dummy",
        "Namespace": "Dummy",
        "Period": 300,
        "Statistic": "Average",
        "Threshold": 0,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "DummyTopic18541D2D0": Object {
      "Type": "AWS::SNS::Topic",
    },
    "DummyTopic2F00C887A": Object {
      "Type": "AWS::SNS::Topic",
    },
    "DummyTopic389AA825B": Object {
      "Type": "AWS::SNS::Topic",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
